lib/Common/Memory/IdleDecommitPageAllocator.cpp

```diff
@@ -96,7 +117,26 @@ IdleDecommitPageAllocator::LeaveIdleDecommit(bool allowTimer)
 #ifdef IDLE_DECOMMIT_ENABLED
     if (allowTimer)
     {
-        cs.Enter();
+        if (!cs.TryEnter())
+        {
+            this->tryEnterIdleDecommit = true;
```

Comment:
A better name for `tryEnterIdelDecommit` would be `waitingToEnterIdleDecommit`.

Reply:
Great suggestion. Renamed.

---
In reply to: [196262994](https://github.com/Microsoft/ChakraCore/pull/5326#discussion_r196262994) [](ancestors = 196262994)